### PR TITLE
Fixes

### LINKED ISSUE #29: Mobile Issue
Hi,
I was hoping that someone else would point it out but I just wanted to let you know in case you weren't already aware, for the mobile version of Yiffspot, pressing the button containing three horizontal bars which previously before the switch to WS would drop down the preferences tab, no longer functions. I have tried multiple mobile browsers, uninstalling them and trying again to no avail. Mobile browsers that open websites in desktop mode do function so I know Yiffspot is working. 
Hope this is a quick fix.

### LINKED ISSUE #39: Mobile window sizing issues
While the responsive layout basically works on mobile, when the chat window expands to the overflow point, it seems that the page layout doesn't know what to do with itself and starts expanding, meaning the top menu scrolls out of the frame and navigation gets awkward. This happens with both stock yiffspot and the teledildonics fork.

I think this could be fixed using viewport units for max-height/width, but this is gonna require the buggyfill package for the fucked up viewport issues on safari. https://github.com/rodneyrehm/viewport-units-buggyfill

### LINKED ISSUE #44: allow links to chatzy
sometimes people like to move to a more stable platform or have long term RPs, please allow links to Chatzy and any other chatsites

### LINKED ISSUE #46: Problem: Chat wont show up after I do certain things
Problem: [Read text above]
Expected outcome: Shows text in chat
**Steps to reproduce**
1. Go to Yiffspot.com
2. Set some preferences (Doesn't matter what you set it to)
3. Click “Save Preferences”
4. Now try to find a partner

**Result**
The chat doesnt show anything

### LINKED ISSUE #48: Stylesheets stopped working.
send to devs
Either my laptop didnt load them or the website.

### LINKED ISSUE #49: Document/Tab title not properly updated
Once you have been connected to a partner for the first time, the tab title flashes the "Partner Connected" status forever, even if the chat ends for any reason, and even if you start to look for a new partner. The title should be maintained according to the current status of your connection/ "Partner Connected" when you are connected to someone, "Disconnected" when a chat has ended, and "Looking for Partner" when you are actively searching.

### LINKED ISSUE #52: Highlight matching fetishes when you are matched with someone
Currently, when you are matched with a partner, it just gives you a long block of text listing all the fetishes they have marked. Of course, this info is useful, but I think it would be even more useful if it highlighted or colored the fetishes which you both share, so that you can more easily see what you have in common at a glance.

Thanks!

### LINKED ISSUE #70: preferences
How to disable some of preferences?

### LINKED ISSUE #72: big issue
There is some big issue.
When i try access chat with no complete perferences, i get notification: "You have attempted to submit invalid preferences. Please check your preferences again."  But even with that notification get connected with users. https://ibb.co/92MVqKy

### LINKED ISSUE #87: Whole Site Broken??
This bug may be a personal thing, but I cannot scroll on the site at all, making it unable to show the "find partner" button while you can do things like zoom the page out to "fix" this problem, there is also an usually low number of active people on the site, which is making me think im not the only one with this problem, side note- mobile works as intended from what i can tell, still low user count

### PR DESCRIPTION
**Pull request by xhayper**
Fix Scrollbar not appearing in the sidebar,
Fix disconnect button not working sometimes,
Fix find partner's logic,
Fix a bug where "Any/All" come back after you reload the page.
Fix color for search bar inside multi-select box.
HOPEFULLY fix a bug where you somehow manage to match up with 2 partners.
Added Gender Fluid.
Added Werewolf.
Fixed a bug with the Block button.
Fix some CSS/Layout.

Fixes #87 
Fixes #72 
Fixes #70 
Close #52
Close #49
Close #48
Close #46 
Close #44
Close #39
Close #29

### CODE REVIEW

**Comment by xhayper** [comment]
@kisuka Last update for today, i promise ;w;

File: src/server/find-partner.js [hunk 1]
```diff
@@ -58,14 +58,14 @@ const matchedPreferences = (user, partner) => {
...
-    ++matchCount;
+    matchCount++;
   }
 
   if ((user.partner['role'] == partner.user['role'] && partner.partner['role'] == user.user['role']) || (user.user['role'] == 'Switch' || partner.user['role'] == 'Switch')) {
     matchCount++;
   }
 
-  return matchCount >= 3;
+  return matchCount >= 4;
```
Comment (NotSafeForWeasels) [inline src/server/find-partner.js:68]: This if statement is duplicated. You should just remove the duplicate code and then the `matchCount` variable will work properly again with just 3.

**Review by NotSafeForWeasels — CHANGES_REQUESTED** [review]
Found a minor issue with the code, a duplicate if that would be best omitted

File: src/server/find-partner.js (hunk 1 above)
Comment (xhayper) [reply src/server/find-partner.js:68]: mb.

**Comment by xhayper** [comment]
Thank you!

**Comment by xhayper** [comment]
All good now.

**Pull request merged**